runtime/minor_gc.c

```diff
@@ -313,6 +313,23 @@ static void oldify_one (void* st_v, value v, volatile value *p)
     st->live_bytes += Bhsize_hd(hd);
     result = alloc_shared(st->domain, sz, tag, Reserved_hd(hd));
     field0 = Field(v, 0);
+    if (tag == Closure_tag) {
+      /* Copy unscannable prefix, which will include all infix tags,
+       * so that any domain can oldify infix pointers to `v` into
+       * pointers which also have Infix_tag to a working header (so
+       * that, e.g., major GC marking can work while the block is
+       * still on our oldify todo list). Have to do this here, before
+       * we update the object header.
```

Comment:
This looks good but I don't understand the last sentence. Why is there an ordering requirement with the object header?